### PR TITLE
Add searching customers by phone number

### PR DESCRIPTION
**Pull request by KonstantinosPapadopoulos**
I want to merge this change because...

# Impact

* [ ] Added option to filter staff by phone number as well - many customers don't remember their order number and saying their email through the phone can be tricky, so searching by phone number (for example in the dashboard 'Orders' section) can be useful.

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by maarcingebala** [comment]
Hey @KonstantinosPapadopoulos, thanks for opening the PR! Would you mind adding a test for these changes? Example test that checks some other filtering on that query is here: https://github.com/mirumee/saleor/blob/70d382759cb0e2a7184c50ff588ffdd0c2e0f862/saleor/graphql/account/tests/test_account.py#L3649

We would need to add another function that tests filtering by phone number.

File: saleor/graphql/account/tests/test_account.py [hunk 1]
```diff
@@ -3656,6 +3660,8 @@ def test_query_customer_members_with_filter_search(
     staff_user,
 ):
 
+    # print("Address is: " + str(address)
+    print("test ==== ")
```
Comment (maarcingebala) [inline saleor/graphql/account/tests/test_account.py:3664]: This should be removed.

**Review by maarcingebala — COMMENTED** [review]
The test looks good, but CI is failing with some checks. To fix it you need to format the code with`black`, `flake8` and `isort`. See the CI checks that failed for more details: https://github.com/mirumee/saleor/pull/6271/checks?check_run_id=1317776749 and let me know if you have any questions!

**Comment by KonstantinosPapadopoulos** [comment]
Thanks a lot @maarcingebala for your help. Hopefully it should be fixed now. I am not very sure why the code coverage is failing though :/

**Review by maarcingebala — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Comment by maarcingebala** [comment]
@KonstantinosPapadopoulos  Thank you for your contribution to Saleor!

**Pull request merged**